test/po_add_format_hints_test.rb

```diff
@@ -0,0 +1,110 @@
+    end
+
+    it "Detects more complex printf-like format strings: %-10.4d" do
+      expect(subject.find_known_formats("%2.1d")).to be == "c-format"
+      expect(subject.find_known_formats("%-10.4d")).to be == "c-format"
+      expect(subject.find_known_formats("%+10.4d")).to be == "c-format"
+      expect(subject.find_known_formats("% 10.4d")).to be == "c-format"
+    end
+
+    it "Detects positional parameters in printf-like format strings: %1$d, %2$s, ..." do
+      expect(subject.find_known_formats("%1$d")).to be == "c-format"
+      expect(subject.find_known_formats("%1$s")).to be == "c-format"
+      expect(subject.find_known_formats("%2$s %1$s")).to be == "c-format"
+    end
+
+    it "Detects sformat-like format strings: %1, %2, ..." do
+      expect(subject.find_known_formats("foo %1 bar")).to be == "ycp-format"
+      expect(subject.find_known_formats("%2 foo %1 bar")).to be == "ycp-format"
+    end
+
+    it "Detects named parameters: %{foo}, %{bar}" do
+      expect(subject.find_known_formats("%{foo} is %{bar}")).to be == "perl-brace-format"
+    end
+
+    it "Detects named parameters with underscore in the name: %{foo_bar}" do
+      expect(subject.find_known_formats("%{foo}")).to be == "perl-brace-format"
+      expect(subject.find_known_formats("%{foo_bar}")).to be == "perl-brace-format"
+    end
+
+    it "Ignores named parameters with format spec: %<foo>s, %<bar>d" do
+      expect(subject.find_known_formats("%<foo>s is %<bar>d")).to be == nil
+    end
+
+    it "Survives inconsistent formats: %1 %2d (c-format wins)" do
+      expect(subject.find_known_formats("%1 %2d")).to be == "c-format"
+      expect(subject.find_known_formats("%1s %2")).to be == "c-format"
+    end
+
+    it "Can handle embedded newlines" do
+      expect(subject.find_known_formats("foo\nbar\n%s bar")).to be == "c-format"
+    end
+
+    it "Can handle escaped percent signs" do
+      expect(subject.find_known_formats("foo 42%% bar")).to be == nil
+      expect(subject.find_known_formats("foo %%1 bar")).to be == nil
+      expect(subject.find_known_formats("foo %%s bar %1")).to be == "ycp-format"
+      expect(subject.find_known_formats("foo %%1 bar %s")).to be == "c-format"
+    end
+
+    it "Can handle unescaped percent signs" do
+      expect(subject.find_known_formats("foo 42% bar")).to be == nil
+      expect(subject.find_known_formats("100% complete: %1")).to be == "ycp-format"
+    end
+
+    it "Ignores named parameters if perl-brace-format is disabled" do
+      subject.perl_brace_format = false
+      expect(subject.find_known_formats("%{foo} is %{bar}")).to be == nil
+      subject.perl_brace_format = true
+      expect(subject.find_known_formats("%{foo} is %{bar}")).to be == "perl-brace-format"
+    end
+  end
+
+  describe "#run" do
+    # Always use "--dry-run" to avoid modifying the sample .pot file!
+    it "Reports the expected number of changes" do
+      argv = ["--dry-run", "data/hello.pot"]
```

Comment:
```suggestion
      argv = ["--dry-run", "#{__dir__}/data/hello.pot"]
```

also 2 times below, so that `rspec test/*test.rb` works
and if you make your PR branch in the main repo the next time I can push my fixes directly

Reply:
Here: https://github.com/yast/yast-devtools/commit/8cfc675074745ab5618db13c8a4555d69e89860d

Reply:
Done (with a constant; DRY etc.)